samples/base/Logger.cpp

```diff
@@ -88,14 +96,37 @@ namespace darwin {
             _name = name;
         }
 
+        bool Logger::setFilePath(std::string const& filepath) {
+            bool ret = true;
+
+            std::ofstream testfile = std::ofstream(filepath, std::ios::out | std::ios::app);
+
+            if (!testfile.is_open()) {
+                std::clog << "Can't open log file " << filepath << std::endl;
+                ret = false;
+            } else {
+                _filepath = filepath;
+                _file.swap(testfile);
+            }
+
+            if (testfile.is_open())
+                testfile.close();
```

Comment:
Useless statement, no? ofstream will be closed when the destructor is called at the end of the scope, no?

Reply:
Seems so, I tend to stick to my (not so) good old C best practices instead of embracing C++ semantics

Reply:
change made, I leave you to comment/resolve the conversation